dns/zone.py

```diff
@@ -1085,3 +1085,45 @@ def from_xfr(xfr, zone_factory=Zone, relativize=True, check_origin=True):
     if check_origin:
         z.check_origin()
     return z
+
+def generate_ptr(zone, relativize=False, zone_factory=Zone):
+    """Generate PTR Zones for a given zone file and returns a dict
+    with origin as key and the zone object as value.
+
+    @param zone: the original zone object
+    @type zone: dns.zone.Zone
+    @param zone_factory: The zone factory to use
+    @type zone_factory: function returning a Zone
+    @raises KeyError: if zone parameter is not a Zone object.
+    @rtype: dict
+    """
+    import dns.reversename
+    import dns.rdtypes.ANY.PTR
+    import copy
+    if not isinstance(zone, Zone):
+            raise KeyError("zone parameter must be a Zone object")
+    # Deep copy to be able to change the encaplusated rdata object without
+    # modifying the original Zone.
+    SOA = copy.deepcopy(zone.find_rdataset(zone.origin,dns.rdatatype.SOA))
+    # As SOA is a singleton type we can simply choose relativity
+    # for the encapsulated rdata object for the first object in the set.
+    SOA.items[0].choose_relativity(origin=zone.origin,relativize=False)
+    ptr_zones = {}
+    for rdt in [dns.rdatatype.A, dns.rdatatype.AAAA]:
```

Comment:
Please use `(dns.rdatatype.A, dns.rdatatype.AAAA)` use rather generators/tuples than lists to safe resources (even if here are resources spent minimal :-))